pkg/quarterdeck/api/v1/options.go

```diff
@@ -13,3 +13,27 @@ func WithClient(client *http.Client) ClientOption {
 		return nil
 	}
 }
+
+func WithCredentials(creds Credentials) ClientOption {
+	return func(c *APIv1) error {
+		c.creds = creds
+		return nil
+	}
+}
+
+// RequestOption allows us to configure individual APIv1 client requests
+// TODO: this is just a hack that modifies the request to get us started, but we will
+// likely want to consider what design pattern we want to use in SC-12797.
+type RequestOption func(req *http.Request) error
+
+// WithRPCCredentials overwrites any existing Authorize header with the specified creds.
+func WithRPCCredentials(creds Credentials) RequestOption {
+	return func(req *http.Request) (err error) {
+		var token string
+		if token, err = creds.AccessToken(); err != nil {
+			return err
+		}
+		req.Header.Set("Authorization", "Bearer "+token)
+		return nil
+	}
+}
```

Comment:
This should work for now, I'm leaning towards the context method but it's hard to tell which way I like better until I actually start implementing it.